DependencyInjection/HttplugExtension.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Http\ClientBundle\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\Loader\XmlFileLoader;
+use Symfony\Component\HttpKernel\DependencyInjection\Extension;
+
+/**
+ * @author David Buchmann <mail@davidbu.ch>
+ */
+class HttplugExtension extends Extension
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function load(array $configs, ContainerBuilder $container)
+    {
+        $configuration = $this->getConfiguration($configs, $container);
+        $config = $this->processConfiguration($configuration, $configs);
+
+        $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+
+        $loader->load('discovery.xml');
+        foreach ($config['classes'] as $service => $class) {
+            if ($class) {
```

Comment:
its the cheapest way to know if the string is non-empty. but i change to
!empty() to make it more explicit.

On 17.10.2015 21:12, Márk Sági-Kazár wrote:
> In DependencyInjection/HttplugExtension.php
> <https://github.com/dbu/HttplugBundle/pull/8#discussion_r42313006>:
> 
>> + */
>> +class HttplugExtension extends Extension
>> +{
>> +    /**
>> +     * {@inheritdoc}
>> +     */
>> +    public function load(array $configs, ContainerBuilder $container)
>> +    {
>> +        $configuration = $this->getConfiguration($configs, $container);
>> +        $config = $this->processConfiguration($configuration, $configs);
>> +
>> +        $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
>> +
>> +        $loader->load('discovery.xml');
>> +        foreach ($config['classes'] as $service => $class) {
>> +            if ($class) {
> 
> What do you check here? Maybe a more explicit check (isset, empty, value
> check)?
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dbu/HttplugBundle/pull/8/files#r42313006>.
> 

-- 
Liip AG // Agile Web Development // T +41 43 500 39 80
CH-8005 Zurich // PGP 0xA581808B // www.liip.ch